ml_tools/buffercache.py

```diff
@@ -0,0 +1,67 @@
+import h5py
+import os
+import numpy as np
+from multiprocessing import Lock
+
+
+class BufferCache:
```

Comment:
How about `FrameCache` or `FrameDiskCache`? "Buffer" and "Cache" kinda mean the same thing.

Reply:
Is it a good idea to put a destructor in here that ensures the file is deleted? or a try finally around clipclassifier code to ensure the same?